parlai/core/tod/teacher_metrics.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Task Oriented Dialogue (TOD) teacher metrics.
+"""
+from typing import Optional, List, Dict, Any
+from parlai.core.metrics import AverageMetric, BleuMetric, F1Metric, Metric, Metrics
+
+
+class SlotMetrics(Metrics):
+    """
+    Helper container which encapsulates standard slot metrics in task oriented learning
+    (jga, slot_p, slot_r, etc).
+
+    Due to differences in dialogue representations between tasks, the input is pre-
+    parsed ground truth and predicted slot dictionaries.
+
+    The 'jga+nlg' metric assumes a balanced set of JGA and NLG scores such that
```

Comment:
Oh hmm - can you explain this a bit more? i.e. when you say `2 * Avg(JGA, NLG_BLEU)`, do you mean the average between JGA and NLG_BLEU, further averaged over the population, i.e. `2 * sum(mean(JGA,NLG_BLEU))/num_samples`? Because if so, wouldn't that just be the same as `2 * sum((JGA+NLG_BLEU)/2)/num_samples`, i.e. `sum(JGA+NLG_BLEU)/num_samples` or `Avg(JGA + NLG_BLEU)`?

Reply:
Ahh this was a copy/pasta of a comment from something that @skiingpacman added. 

But yeah, the intent of the jga+nlg metric is to get an equal weighing between the two. In hindsight, for the purposes of open sourcing, especially since `jga+nlg` isn't being used anywhere, I'll just remove this.